### DISCUSSION TITLE
Boltzmann is a motherfucker

### DISCUSSION THREAD

**Discussion by ta-rekt** [josePereiro/TankeFactory#4] [Discussions]
The discussion right after equation 20 is extremely deep. I see why one should maximize the number of configurations the system can be in in order to give 'the most honest description possible' of the system. But it breaks my brain to try to understand why *nature* also behaves in that way as well. Why should it? I mean, of course it doesn't always, right, if we are missing information in our prior. But once we have 'sufficient' information, maximum entropy gives a good description of what nature does. This puzzles me very much.

**Comment by josePereiro** [comment]
ALERT, LONG REPLY, BUT MAYBE NOT AN ANSWER: If I have a die, I can characterize it, to measure the relative frequencies of each possible outcome in a large number of rolls. Additionally, I can do the same with a lot of dice. Now, given that I know the statistical characteristics of the dice, and the independence between the rolls, I can reason to get the statistics of simultaneous rolling of all dice. As you know, this statistic is very particular, in the sense that the relative variance of important observables vanish as we increase the number of dice used. Actually, for the typical case, there are important questions that we can answer with practical certainty for, let say, 1000 dice. There are A LOT of information about the dice and the rolling process that we ignore, but, knowing the independent dice statistics is 'sufficient' for defining those observables, and not only that, it is the minimal amount of information required for doing so. That is one of the points Jaynes support, after we define our initial information, irrespective of the source (measurements, physics, intuition, etc), the rest is just reasoning, logic. If you believe in the former, you must believe in the later, otherwise you are an irrational being. 

Now, you are asking me about, why does nature seems to care about being reasonable? I don't know. This is a similar question we discussed in Cuba related with [Galileo's thought experiment](https://en.wikipedia.org/wiki/Galileo%27s_Leaning_Tower_of_Pisa_experiment). I see it now just as another empirical law. If you trust a set of premises, but nature fails to behave according to its reasonable consequences, then you must re-evaluate your premises. Again and again, humanity have found that, in fact, a new set of premises can be found that both, we trust, and nature respect in a larger extent. That is, Science is a gradient descent to find the optimal set of premises, and we are the computers.

**Comment by josePereiro** [comment]
NOW AN ANSWER: if I understand correctly, you are curious about why nature appears to prefer the distribution that maximizes entropy from all available distributions, plus the very weird fact that such distribution is also very meaningful from an information theory point of view. 

I think we can reconcile the two views by the fact that the MaxEnt distribution is also the distribution which maximizes the likelihood for the available information (information = constraints,  priors, etc...). Note that in the case of thermodynamics, a very important data/constraint, is that we consider to be very inverosimil, for instance, that particles are in an "special" microstate. We assign uniform priors as much as we can. This is important because then, it is only reasonable to describe the system by using the MaxEnt distribution. It is the more probable given such premises. 

In the evolution of thermodynamic I believe the reasoning is in zig-zag. At the time of Boltzmann, I think they were already aware of the homogeneous nature of gasses and its implications once you consider the atomic nature of it composition. At the same time, macroscopic thermodynamics was already mature enough so the functional nature of the entropy was well described for gases too. Boltzmann joined both stuff, the reasonable premises of microscopic probability uniformity, and the fact that the most probable distribution lead to a function that behave like the classical entropy. 

Once again, If you don't trust the maximum entropy distribution, you must revise the premises, or be irrational, and select a less probable distribution.

**Reply by ta-rekt** [reply]
I think I get it now. MaxEnt doesn't describe how nature behaves. It is just the best we can do if we want to be rational beings, as you say. If the maxent distribution doesn't correspond to what we observe, then we must revise our premises, as Jaynes says later in the paper.

The thing that was bothering me was this. If you make an experiment on some system and then measure it, the distribution of states you observe depends on how the experiment and measurement were done. So why does maximizing entropy subject to the "information" we have, on its own, give a good a priori description of what's happening? because that information contains also everything about the experiment, namely what the experiment keeps constant across measurements. For example you measure a gas at constant energy, maximum entropy will give you the Boltzmann distribution, which is a good description of a gas at fixed energy. But if you measure it after doing work on it or letting it dissipate heat into a reservoir, in between each measurement, it will no longer have that distribution. So when we say "subject to the information we have", this means, taking into account what is kept fixed across measurements.

**Reply by ta-rekt** [reply]
but there may be variables that are not kept fixed across measurements, but are known. then these variables are parameters of the experiment.

**Reply by josePereiro** [reply]
I don't remember if I talked with you about it already, but, It appears to me that there is a clear relationship between the average cost of acquiring the minimal information required for a meaningful description of a system under study, and the culture of the field at hand. For instance, in Physics, a lot can be said from what we can measure 'easily', so, Physicist are modeling/string-inference/deductive minded scientist. In contrast, Biology has being less "modelable". The minimal (sufficient) information required for meaningful inference is just already a lot, so, the field is more about experimental exploration than theoretical modeling. But this is just luck. It could have been that the movement of the planets required 10^23 highly interactive degrees of freedom to be described, so, no Kepler, No Newton,... In the extreme case where we don't have 'easy' information for doing any sophisticated inference, probably not even science would be developed. It is like, the fitness landscape of science is, for our pleasure, sufficiency smooth.

**Reply by ta-rekt** [reply]
Yes 100%. and mathematicians and theoretical computer scientists exist at the boundary of hardness what we as humans can deal with, and slowly push it forward. they are only constrained by the difficulty of reasoning about things.

**Comment by ta-rekt** [comment]
Summarizing what we talked about at ENS at 3 am last week: you sit down and figure out what is fixed about your system, say energy, and you find the maxent distribution under this constraint. Then you get up and do an experiment and find that the distribution of states follows what your maxent predicts. Why should the world behave according to how you happen to set your constraints? The zeroth order answer is if the world may not behave as you expect, if you set the wrong constraints.
But there is another aspect, which is that this depends on how you do your experiment. Here we're supposing an experiment is a set of I.I.D samples of your system. Maxent will agree with your experiment if what is "fixed" and what is "random" are the same in both your model and your experiment. Then it's only a matter of logic, as Jaynes argues, that they should agree. It's not something deep about the world, it's something deep about information theory.

**Reply by josePereiro** [reply]
Mechanics (like Newton's) is an abstraction that aims (mostly successfully) to include all that is relevant for explaining a system, which is why there are no random components by default. Classical thermodynamics (pre-atomic theory) is similar; the relevant quantities follow deterministic laws. The problem arises when bridging the two. Newton's mechanics imply the existence of 'moles of parameters,' and we can't know them all. However, classical thermodynamics shows us that there are deterministic laws (abstractions) that successfully describe features of the system without explicitly accounting for those parameters. But how? How is it possible that most of the information about the system (the precise mechanical state) is so irrelevant to the thermodynamic picture?

That's where Boltzmann proposed/found an interesting connection, unexpectedly, he introduced a probability framework. By assuming each microstate as equiprobable, classical thermodynamics can be recast as a description of the relations between classes/groups/populations of microstates (macrostates). These macrostates are found by performing ensemble averages, and it turns out that 'something deep about ~information~ probability theory' causes the most probable macrostates (equilibrium) to behave exactly as classical thermodynamics prescribes. The explanation for why this is so successful, I think, is still not fully established. $S=k\ln\Omega$ is taken as a new axiom that bridges Mechanics and Thermodynamics. 

What Jaynes proposed is NOT a new way to explain why $S=k\ln\Omega$ is successful. Jaynes discovered a larger pattern (an epistemological one) after Shannon proposed a very compelling measure of information. This pattern suggests that the maximization of entropy always results in the least biased representation of the knowledge you are modeling. Now, why and how information is related to thermodynamics through $S=k\ln\Omega$ and the second law? To my knowledge (not a solid one), it is not yet established. At a minimum, we need to explain why $S=k\ln\Omega$ works in the first place (for instance, what is the mechanical meaning that each microstate is equiprobable).
